utils/pg-introspection/src/acl.ts

```diff
@@ -134,80 +134,67 @@ export type ResolvedPermissions = Omit<AclObject, "role" | "granter">;
 const parseIdentifier = (str: string): string =>
   str.startsWith('"') ? str.replace(/"("?)/g, "$1") : str;
 
+// https://www.postgresql.org/docs/current/ddl-priv.html#PRIVILEGE-ABBREVS-TABLE
+const ACL_MAP = {
+  r: 'select',
+  w: 'update',
+  a: 'insert',
+  d: 'delete',
+  D: 'truncate',
+  x: 'references',
+  t: 'trigger',
+  X: 'execute',
+  U: 'usage',
+  C: 'create',
+  c: 'connect',
+  T: 'temporary',
+  m: 'maintain',
+};
+const NO_PERMISSIONS: AclObject = Object.values(ACL_MAP).reduce((acc, val) => {
+  acc[val] = acc[val + "Grant"] = false;
+  return acc;
+}, { role: "public", granter: "" });
+
 /**
  * Accepts an ACL string such as `foo=arwdDxt/bar` and converts it into
  * a parsed AclObject.
  */
 export function parseAcl(aclString: string): AclObject {
-  // https://www.postgresql.org/docs/current/ddl-priv.html#PRIVILEGE-ABBREVS-TABLE
-
-  const matches = aclString.match(/^([^=]*)=([rwadDxtXUCcTm*]*)\/([^=]+)$/);
-
-  if (!matches) {
+  const acl = Object.assign({}, NO_PERMISSIONS);
+  const roleEndIndex = aclString.indexOf("=");
+  if (roleEndIndex === -1) {
     throw new Error(`Could not parse ACL string '${aclString}'`);
+  } else if (roleEndIndex > 0) {
+    acl.role = parseIdentifier(aclString.substring(0, roleEndIndex));
+  }
+  const aclLength = aclString.length;
+  const lastGrantableTokenIndex = aclLength - 1;
+  let currentPerm: string;
+  let i = roleEndIndex + 1; // skip past the "="
+  // Process the ACL tokens
+  while (i < aclString.length) {
+    const nextChar = aclString[i];
+    if (nextChar === "/") {
+      // granter begins
+      i++;   // skip past the "/" delimiter
+      break; // we're done with ACL tokens
+    }
+    currentPerm = ACL_MAP[nextChar];
+    if (currentPerm === undefined) {
+      throw new Error(`Could not parse ACL string '${aclString}'`);
+    }
+    acl[currentPerm] = true;
+    if (i < lastGrantableTokenIndex && aclString[i + 1] === '*') {
+      // permission + grant
+      i++; // skip past the "*" character
+      acl[currentPerm + "Grant"] = true;
+    }
+    i++;
+  } // end token processing
+  if (i < aclLength) {
```

Comment:
Previously the granter was required, and in general I think it will always be set. Let's just throw an error if it's the empty string.